bundles/org.openhab.binding.boschshc/src/test/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryServiceTest.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.discovery;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openhab.binding.boschshc.internal.devices.BoschSHCBindingConstants;
+import org.openhab.binding.boschshc.internal.devices.bridge.BridgeHandler;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Device;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Room;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ThingUID;
+
+/**
+ * ThingDiscoveryService Tester.
+ *
+ * @author Gerd Zanker - Initial contribution
+ */
+@NonNullByDefault
+public class ThingDiscoveryServiceTest extends ThingDiscoveryService {
+
+    private int thingDiscoveredCalls = 0;
+    @Nullable
+    private DiscoveryResult lastDiscoveryResult = null;
+
+    @Override
+    protected void thingDiscovered(DiscoveryResult discoveryResult) {
+        thingDiscoveredCalls++;
+        lastDiscoveryResult = discoveryResult;
+        super.thingDiscovered(discoveryResult);
+    }
+
+    @Nullable
+    private ThingDiscoveryServiceTest fixture;
+
+    @Nullable
+    private BridgeHandler bridgeHandler;
+
+    @BeforeEach
+    void beforeEach() {
+        fixture = new ThingDiscoveryServiceTest();
+        bridgeHandler = mock(BridgeHandler.class);
+    }
+
+    private void setMockBridgeAsThingHandler() {
```

Comment:
you could consider calling this in beforeEach() always, instead of calling it in many tests (but both is actually fine)